interfaces/sorting.go

```diff
@@ -69,21 +69,27 @@ type byPlugSnapAndName []*snap.PlugInfo
 func (c byPlugSnapAndName) Len() int      { return len(c) }
 func (c byPlugSnapAndName) Swap(i, j int) { c[i], c[j] = c[j], c[i] }
 func (c byPlugSnapAndName) Less(i, j int) bool {
-	if c[i].Snap.InstanceName() != c[j].Snap.InstanceName() {
-		return c[i].Snap.InstanceName() < c[j].Snap.InstanceName()
+	if c[i].Snap.SnapName() != c[j].Snap.SnapName() {
+		return c[i].Snap.SnapName() < c[j].Snap.SnapName()
 	}
-	return c[i].Name < c[j].Name
+	if c[i].Name != c[j].Name {
+		return c[i].Name < c[j].Name
+	}
+	return c[i].Snap.InstanceKey < c[j].Snap.InstanceKey
```

Comment:
Shouldn't we sort by snap name, then instance key and then plug name to be in line with what we did before instances were introduced?

Reply:
Current code orders like this: `foo:a`, `foo_b:a`, `foo:b`. Your proposal would order like this: `foo:a`, `foo:b`, `foo_b:a`. Either works for me, though I find the first variant bit easier to reason about.

Reply:
Now that I read it again I see it makes sense. `foo` & `foo_bar` are different snaps after all.